### DISCUSSION TITLE
trying to setup home assistant vm

### DISCUSSION THREAD

**Discussion by kinghat** [cockpit-project/cockpit-machines#486] [Q&A] [answered]
home assistant distribute a qcow2 for people to setup their vm with but they suggest you use "Generic Linux" as the os variant but i dont see how i can do that in cockpit.

also, it needs to have a uefi firmware but when importing a vm that is not an option, even if i defer immediately starting that vm. is there a way around this?

**Comment by garrett** [comment]
I've opened an issue @ https://github.com/cockpit-project/cockpit-machines/issues/500 for "Generic Linux". I know we use [libosinfo](https://gitlab.com/libosinfo/libosinfo). There's an "unknown" type, but it's filtered out.

Home Assistant is using buildroot and isn't a typical OS https://developers.home-assistant.io/docs/operating-system ... but I'd suggest trying Debian or Alpine for now and see what happens.

**Reply by garrett** [reply]
Oh, as stated in the linked issue, we do actually have generic Linux in the listing.

Make sure you're not selecting download, but using the image via import, as it's a predefined qcow2 image.

![Screenshot 2022-01-04 at 11-12-49 Virtual Machines - garrett Bolt](https://user-images.githubusercontent.com/10246/148043758-e2bc6dda-32ec-45ec-9d4e-9cfccd7f1001.png)

Make sure the qcow2 exists on the system Cockpit is running on, specify the path to the qcow2 file, and you should hopefully be good.

**Comment by garrett** [comment, marked as answer]
> also, it needs to have a uefi firmware but when importing a vm that is not an option, even if i defer immediately starting that vm. is there a way around this?

I'm guessing it should detect a UEFI partition on the disk... but this is a wild guess and there's a good chance I'm just wrong. :wink:

@KKoukiou, @martinpitt, or @marusak would likely know more about UEFI in Cockpit Machines.

**Reply by martinpitt** [reply]
Often it should be auto-detected by libvirt, but if you want to create a new VM, then uncheck the "immediately start" checkbox, and before starting, click on "Firmware":

![image](https://user-images.githubusercontent.com/200109/151936754-1bdfde50-49b9-454d-abe9-7a8cd8df5ed5.png)

and select UEFI:

![image](https://user-images.githubusercontent.com/200109/151936784-5d4f0174-14ba-4589-a7ed-19c1343a3e46.png)
